partcad/src/partcad/part_factory_brep.py

```
@@ -0,0 +1,135 @@
+    def _load_brep_directly(self):
+        """
+        Load a BREP file directly in the main thread.
+        """
+        time.sleep(0.0001)  # Brief pause for thread synchronization
+        try:
+            shape = TopoDS_Shape()
+            builder = BRep_Builder()
+            brep_tools = BRepTools()
+
+            if not brep_tools.Read_s(shape, self.path, builder):
+                raise FileReadError(f"Failed to load BREP file: {self.path}")
```

Ruff (0.8.2)

Abstract `raise` to an inner function

(TRY301)